src/iguana/algorithms/clas12/FiducialFilter/Algorithm.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include "iguana/algorithms/Algorithm.h"
+
+namespace iguana::clas12 {
+
+  /// @brief_algo Filter the `REC::Particle` bank by applying DC (drift chamber) fiducial cuts
+  ///
+  /// @begin_doc_algo{clas12::FiducialFilter | Filter}
+  /// @input_banks{REC::Particle, REC::Traj, RUN::config}
+  /// @output_banks{REC::Particle}
+  /// @end_doc
+  ///
+  /// @begin_doc_config
+  /// @config_param{pass | int | cook type to use for assigning fiducial cuts}
+  /// @end_doc
+  class FiducialFilter : public Algorithm
+  {
+
+      DEFINE_IGUANA_ALGORITHM(FiducialFilter, clas12::FiducialFilter)
+
+    public:
+
+      void Start(hipo::banklist& banks) override;
+      void Run(hipo::banklist& banks) const override;
+      void Stop() override;
+
+      struct traj_row_data {
+        double x1 = -999;
+        double x2 = -999;
+        double x3 = -999;
+        double y1 = -999;
+        double y2 = -999;
+        double y3 = -999;
+        double z1 = -999;
+        double z2 = -999;
+        double z3 = -999;
+        int sector= -1;
+      };
```

Comment:
You either have to document _every_ member here, or tell `doxygen` to ignore it with `@cond`. Here's how you can just have it document the `struct` and ignore the members:
```suggestion
      /// structure to hold `REC::Traj` data
      struct traj_row_data {
        /// @cond NO_DOC
        double x1 = -999;
        double x2 = -999;
        double x3 = -999;
        double y1 = -999;
        double y2 = -999;
        double y3 = -999;
        double z1 = -999;
        double z2 = -999;
        double z3 = -999;
        int sector= -1;
        /// @endcond
      };
```